Source/WebCore/platform/graphics/cg/PathCG.cpp

```diff
@@ -130,16 +164,38 @@ void PathCG::add(PathEllipse ellipse)
     CGPathAddArc(ensureMutablePlatformPath(), &cgTransform, 0, 0, 1, ellipse.startAngle, ellipse.endAngle, ellipse.direction == RotationDirection::Counterclockwise);
 }
 
+static inline void addToCGContextPath(CGContextRef context, PathEllipse segment)
+{
+    CGAffineTransform oldTransform = CGContextGetCTM(context);
+    AffineTransform transform;
+    transform.translate(segment.center.x(), segment.center.y()).rotate(rad2deg(segment.rotation)).scale(segment.radiusX, segment.radiusY);
+    CGAffineTransform cgTransform = transform;
+    CGContextConcatCTM(context, cgTransform);
+    // CG coordinates system increases the angle in the anticlockwise direction.
+    CGContextAddArc(context, 0, 0, 1, segment.startAngle, segment.endAngle, segment.direction == RotationDirection::Counterclockwise);
+    CGContextSetCTM(context, oldTransform);
```

Comment:
Should we use similar technique like what you did for the rounded rect? I think using PathEllipse (not PathEllipseInRect) is not very common.

```
auto path = PathCG::create();
path->add(WTFMove(segment));
```

Reply:
I think with the reused transform code, maybe it's not that complicated to have it duplicated?